frontend/src/app/password-reset/password-reset-confirm.component.html

```diff
@@ -0,0 +1,51 @@
+<div class="container">
+  <h2>Potwierdź zmianę hasła</h2>
+  <form (ngSubmit)="onSubmit()" [formGroup]="resetForm">
+    <div  formGroupName="passwords">
+      <div class="form-group">
+        <label for="password">Podaj nowe hasło</label>
+        <input
+          type="password"
+          id="password"
+          formControlName="password"
+          class="form-control form-control-lg"
+          [ngClass]="{'is-invalid': resetForm.get('passwords.password').invalid && resetForm.get('passwords.password').touched}"
+          />
+        <div
+          class="invalid-feedback"
+          *ngIf="resetForm.get('passwords.password').invalid && resetForm.get('passwords.password').touched">
+          Pole nie może być puste.
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="confirm-password">Powtórz hasło</label>
+        <input
+          type="password"
+          id="confirm-password"
+          formControlName="confirmPassword"
+          class="form-control form-control-lg"
+          [ngClass]="{'is-invalid': resetForm.get('passwords.confirmPassword').invalid && resetForm.get('passwords.confirmPassword').touched}"
+          />
+        <div
+          class="invalid-feedback"
+          *ngIf="resetForm.get('passwords.confirmPassword').invalid && resetForm.get('passwords.confirmPassword').touched">
+          Pole nie może być puste.
+        </div>
+      </div>
+    </div>
+    <div *ngIf="resetForm.get('passwords.confirmPassword').dirty && resetForm.get('passwords').errors?.notEqual">
```

Comment:
Move this logic to component method, e.g.

`arePasswordsEqual() : boolean {
return resetForm.get('passwords.confirmPassword').dirty && resetForm.get('passwords').errors?.notEqual;
}`

Reply:
I know that @juliaanholcer used week ago expression `template-based form` as I unserstand your comments here @CorneeSean, maybe there's an aproach to keet this logic here,

Can You, @juliaanholcer , provide some documentation regarding this `tempalate-based form`?

Reply:
I don't know how it works exactly (I'm not Angular expert), but cannot we use here `checkPasswords` from Component?

Reply:
I've changed template driven form into reactive form.  This `checkPasswords` custom validator returns the error for passwords group and I use this error to determine whether an alert should be displayed or not.
@CorneeSean this method in your example with this logic should be named `arePasswordsNotEqual()` or I can change its logic but then I will use it with negation in the template.